### PR TITLE
update name of component sample

### PR DESCRIPTION
**Pull request by danewalton**
- Updates the name of the component sample since it uses pnp client APIs. The other sample will be updated later once it is updated to use pnp APIs

### CODE REVIEW

File: sdk/samples/iot/iot_sample_common.c [hunk 1]
```diff
@@ -149,7 +149,7 @@ void iot_sample_read_environment_variables(
     {
       case PAHO_IOT_HUB_C2D_SAMPLE:
       case PAHO_IOT_HUB_METHODS_SAMPLE:
-      case PAHO_IOT_HUB_PNP_COMPONENT_SAMPLE:
+      case PAHO_IOT_PNP_COMPONENT_SAMPLE:
       case PAHO_IOT_HUB_PNP_SAMPLE:
```
Comment (momuno) [inline sdk/samples/iot/iot_sample_common.c:153]: Can you move this to be alphabetical like you do in the enum please?

File: sdk/samples/iot/README.md [hunk 2]
```diff
@@ -328,9 +328,9 @@ This section provides an overview of the different samples available to run and
 
 ### IoT Hub Plug and Play Multiple Component Sample
```
Comment (momuno) [inline sdk/samples/iot/README.md:330]: Are we still calling it IoT Hub Plug and Play or IoT Plug and Play?

File: sdk/samples/iot/README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ To run the samples, ensure you have the following programs and tools installed o
 
 - Have the most recent version of [Azure IoT Explorer](https://github.com/Azure/azure-iot-explorer/releases) installed (more instructions can be found [here](https://docs.microsoft.com/azure/iot-pnp/howto-use-iot-explorer)) and connected to your Azure IoT Hub if running a Plug and Play sample:
 
-  *Executables:* `paho_iot_hub_pnp_sample`, `paho_iot_hub_pnp_component_sample`
+  *Executables:* `paho_iot_hub_pnp_sample`, `paho_iot_pnp_component_sample`
```
Comment (momuno) [inline sdk/samples/iot/README.md:59]: will `paho_iot_hub_pnp_sample` also get updated at somepoint?  Why the difference between it and `paho_iot_pnp_component_sample`?

**Review by momuno — APPROVED** [review]

File: sdk/samples/iot/README.md (hunk 2 above)
Comment (danewalton) [reply sdk/samples/iot/README.md:330]: nope good catch

File: sdk/samples/iot/README.md (hunk 3 above)
Comment (danewalton) [reply sdk/samples/iot/README.md:59]: Yup tried to give reasoning above in the pr comment but yes, once the sample is updated to use the pnp api

Comment (momuno) [reply sdk/samples/iot/README.md:59]: you did. that's just me not reading. smh.

**Pull request merged**